packages/block-library/src/post-template/index.php

```diff
@@ -160,3 +160,28 @@ function register_block_core_post_template() {
 	);
 }
 add_action( 'init', 'register_block_core_post_template' );
+
+/**
+ * Adds the post classes to the REST API response.
+ *
+ * @since 6.6.0?
+ */
+function add_post_class_to_api( $data, $post, $context ) {
+	$data->data['post_class'] = get_post_class( '', $post->ID );
+
+	return $data;
+}
+
+/**
+ * Adds the post classes to all post types in the REST API.
+ *
+ * @since 6.6.0?
+ */
+function add_post_class_to_all_post_types() {
+	$post_types = get_post_types( array( 'public' => true ), 'names' );
+
+	foreach ( $post_types as $post_type ) {
+		add_filter( "rest_prepare_{$post_type}", 'add_post_class_to_api', 10, 3 );
+	}
+}
+add_action( 'rest_api_init', 'add_post_class_to_all_post_types' );
```

Comment:
Thanks for the PR! Despite the fact that this code should live in different file(`rest-api.php`), I think calling `get_post_class` is the only reliable way to pass the proper classes in the editor.

I'm wondering though if this could have impact on the REST API and whether it could be added conditionally.. @TimothyBJacobs @spacedmonkey

Reply:
@ntsekouras thanks for helping, I moved the REST API functions to rest-api.php

Reply:
Let's wait for the REST API feedback, but it should be moved in `rest-api.php` under `wordpress-6.6` folder. Sorry for not clarifying.

Reply:
@youknowriad I believe you also had an opinion on this. The alternative is to re-create all the classes in JS.

Reply:
> The alternative is to re-create all the classes in JS.

This won't work with the post_class filter, I think? Being able to add custom post classes is mainly why I made this pull request.

Reply:
> I'm wondering though if this could have impact on the REST API and whether it could be added conditionally.. 

@ntsekouras, do you mean adding the post classes to the REST API only when in the block editor, like this?
```php
if ( get_current_screen()->is_block_editor() ) {
    // Add post classes to REST API
}
```

Reply:
I was thinking if we could consider a different endpoint just for that or a way to add this only if explicitly requested as a field. Otherwise it will be calculated/fetched for every request we make for a post and there is not something needed in general. 

I [pinged core-restapi ](https://wordpress.slack.com/archives/C02RQC26G/p1713938610217419)channel to try to get some feedback.

Reply:
No it, wouldn't work with the PHP filter, but we could add a similar filter in JS. Just trying to thing about the alternatives here.

Reply:
>  Otherwise it will be calculated/fetched for every request we make for a post and there is not something needed in general.

I'm not particularly concerned with this. Looking at the function, the performance impact appears to be pretty minimal. And just because a client might not need all the fields in a response, doesn't mean we don't include them by default. Clients that are concerned about performance can always use `_fields` to limit the fields to the ones they need.

Reply:
> No it, wouldn't work with the PHP filter, but we could add a similar filter in JS. Just trying to thing about the alternatives here.

Yes that's true, recreating all the classes in JS along with a JS filter could be an alternative solution.